### PR TITLE
Fix array and memory segment access in VectorAPIExpansion

### PR DESCRIPTION
**Pull request by gita-omr**
- we should not add array header to the address but use base address and offset provided by the intrinsic, whether it's array or segment

### CODE REVIEW

**Review by tajila — APPROVED** [review]
The approach makes sense to me. @0xdaryl Please review this

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp [hunk 1]
```diff
@@ -1267,6 +1267,13 @@ TR_VectorAPIExpansion::generateAddressNode(TR::Compilation *comp, TR::Node *arra
    return aladdNode;
    }
 
+TR::Node *
+TR_VectorAPIExpansion::generateAddressNode(TR::Compilation *comp, TR::Node *base, TR::Node *offset)
```
Comment (0xdaryl) [inline runtime/compiler/optimizer/VectorAPIExpansion.cpp:1271]: The compilation object is not used in this function.  Does it need to be part of the API?

Comment (gita-omr) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp:1271]: Right. I will remove it.

Comment (gita-omr) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp]: Done.

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk21

**Comment by gita-omr** [comment]
jdk_utik_0 failure on Z seems to be unrelated.

**Comment by gita-omr** [comment]
Squashed. Thanks!

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**